src/Targets/Microsoft.Managed.DesignTime.targets

```diff
@@ -29,6 +29,7 @@
     <ProjectCapability Include="ManagedLang"/> <!-- Temporary: See https://github.com/dotnet/roslyn-project-system/issues/47 -->
     <ProjectCapability Include="HandlesOwnReload"/>
     <ProjectCapability Include="UseFileGlobs"/>
+    <ProjectCapability Include="ManagedEditAndContinue"/>
```

Comment:
Shared Project inherit their language service from projects that are referencing it. Your component *isn't going to work* because there isn't a host to delegate onto. 

Can you see if the behavior (minus your component) for shared projects matches the existing desktop behavior? If so, file a bug. Otherwise, investigate why the difference.